packages/wallets/src/components/WalletTourGuide/WalletTourGuide.tsx

```diff
@@ -51,24 +67,42 @@ const WalletTourGuide = () => {
 
         if (action === ACTIONS.RESET) {
             setWalletsOnboarding('');
+            setRun(false);
             if (!isAllWalletsAlreadyAdded && addMoreWalletRef.current) {
                 addMoreWalletRef.current.style.transform = addMoreWalletsTransformValue;
             }
         }
     };
 
     useEffect(() => {
-        const switchToFiatWallet = () => {
+        const needToStart = walletsOnboarding === START_VALUE;
+        if (needToStart) {
+            const isAnythingLoading =
+                dxtradeIsLoading ||
+                ctraderIsLoading ||
+                isFetching ||
+                !isSuccess ||
+                isLoading ||
+                sorteAccountsListLoading;
             if (fiatWalletLoginId && fiatWalletLoginId !== activeWalletLoginId) {
                 switchWalletAccount(fiatWalletLoginId);
+            } else if (needToStart && !isAnythingLoading) {
```

Comment:
@wojciech-deriv pls check the sonarcloud issue related to `needToStart`

Reply:
updated PR with slightly different solution, and from what I see, sonar is clear,